extension/source/contentScript.ts

```diff
@@ -0,0 +1,139 @@
+/**
+ * The only thing this does is inject the in-page script and relay communication
+ * between that in-page script and the background script.
+ *
+ * Why do we need to do it that way? That's a good question.
+ *
+ * The in-page script cannot talk directly to the background script. This is
+ * not our design, it's a limitation of how WebExtensions work which is
+ * motivated by security. The content script is trustable because it's an
+ * independent JavaScript context. This means the dApp cannot interfere with
+ * what the content script does. E.g. If the dApp does any monkeypatching or
+ * prototype pollution, the content script will not be affected.
+ *
+ * We do make use of this trusted context feature by setting the origin and
+ * providerId here, so that the background script can safely decide which
+ * information is appropriate for its responses using these fields.
+ *
+ * However, the browser could easily provide this kind of information securely
+ * as part of an api that allows direct communication between the in-page script
+ * and the background script. If that ever happens, we should most likely use
+ * that and delete the content script entirely.
+ *
+ *      +----------------------------------------------------------+
+ *      |                     Background Script                    |
+ *      +----------------------------------------------------------+
+ *              |                    |                    |
+ *      +----------------+   +----------------+   +----------------+
+ *      | Content Script |   | Content Script |   | Content Script |
+ *      +----------------+   +----------------+   +----------------+
+ *              |                    |                    |
+ *      +----------------+   +----------------+   +----------------+
+ *      |      dApp      |   |      dApp      |   |      dApp      |
+ *      | In-Page Script |   | In-Page Script |   | In-Page Script |
+ *      +----------------+   +----------------+   +----------------+
+ */
+
+import { runtime } from 'webextension-polyfill';
+import assertType from './cells/assertType';
+import isType from './cells/isType';
+import RandomId from './helpers/RandomId';
+import { RpcRequest, RpcResponse, RpcResult, toRpcResult } from './types/Rpc';
+
+(() => {
+  if (
+    document.doctype?.name !== 'html' ||
+    location.pathname.endsWith('.pdf') ||
+    document.documentElement.nodeName.toLowerCase() !== 'html'
```

Comment:
Probably not an issue, but any chance we could load an `.htm` page?

Also, why is the `.pdf` blocking needed?

Would it be better to do a MIME type check for these, assuming a way to check that exists on `document`, `location`, or somewhere else in a web API.